### PR TITLE
Closes #907 - Rewrite the README.release.md to a template for a task list #907

### LINKED ISSUE #907: Rewrite the README.release.md to a template for a task list
Rewrite [README.release.md](https://github.com/nmstate/nmstate/blob/master/packaging/README.release.md) to a [task list](https://help.github.com/en/github/managing-your-work-on-github/about-task-lists)

Ideally, this would be a template for new issues.

### PR DESCRIPTION
**Pull request by pavitra14**
# What changed
Converted the Lists to Checkboxes/Task List
Closes #970
# Before
before-gh

# After
after-gh

### CODE REVIEW

**Comment by tyll** [comment]
Thank you, this looks nice. A nice bonus would be to look into https://help.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository and make it an actual template that could be selected when creating new issues. You will have to test this on your fork and then use the generated files to create the PR for this repo. The `README.release.md` should then only contain some reference to using the tamplate to create a new issue and a reference where to find the template.

**Comment by pavitra14** [comment]
@tyll Just confirming if I understood correctly,
So I will need to create a new PR with the template and Add more commits to this PR for this readme to contain just references to the new template.

**Comment by tyll** [comment]
> @tyll Just confirming if I understood correctly,
> So I will need to create a new PR with the template and Add more commits to this PR for this readme to contain just references to the new template.

@pavitra14 you would need to 

1. go to your nmstate for at https://github.com/pavitra14/nmstate
2. follow the instructions at https://help.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository to setup an issue template using the new README that you added in this PR
3. Get the template from your fork and change this PR to store the template at the right place
4. Update the README to mention the issue template

**Comment by ffmancera** [comment]
We are trying this document format in #916. I will give feedback :-)

**Comment by tyll** [comment]
The section at https://github.com/nmstate/nmstate/pull/912/files#diff-a239ee76e414afea86b41ea5d9a8b8efR3 about creating Jira tickets for missing steps can be removed now, since the task list is used to track progress. 

Also it seems a checkbox for creating the changelog is missing.

File: packaging/README.release.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ an entry like the following:
 
 ## Tagging
```
Comment (tyll) [inline packaging/README.release.md]: Please also add a checkbox for the changelog creation.

File: packaging/README.release.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ an entry like the following:
...
 
-* Tag new release in git.
+- [ ] Tag new release in git.
```
Comment (tyll) [inline packaging/README.release.md]:
```suggestion
- [ ] Tag new release in git.
- [ ] Ensure that the vdsm tests ran against this state
```

File: packaging/README.release.md (hunk 1 above)
Comment (pavitra14) [reply packaging/README.release.md]: Added, will push in the next commit.

**Comment by tyll** [comment]
When you are done, please squash the commits together (check the git doc if you need help) and adjust the commit message to follow https://github.com/nmstate/nmstate/blob/master/commit-template.txt

**Comment by pavitra14** [comment]
I tried rebasing to squash the commits, it seems like there was a issue with the head :/

**Comment by tyll** [comment]
> I tried rebasing to squash the commits, it seems like there was a issue with the head :/
> Let me rebase the HEAD from start

What is your git setup/problem? Usually it should work like this:

```
git fetch upstream
git rebase -i upstream/master
```
For this you need to have added the upstream repo as upstream remote.

**Comment by pavitra14** [comment]
Yes I did the same.
My git version : 
```
git version 2.21.1 (Apple Git-122.3)
```
You can squash the commits while merging.

**Comment by tyll** [comment]
> Yes I did the same.
> My git version :
> 
> ```
> git version 2.21.1 (Apple Git-122.3)
> ```
> 
> You can squash the commits while merging.

Good. There are two commits in this PR, please squash them, so that there is only one commit. Also please reword the commit message to follow the template I linked.

File: packaging/README.release.md [hunk 3]
```diff
@@ -1,11 +1,10 @@
 # Creating a New Release
```
Comment (tyll) [inline packaging/README.release.md]: Please change this file to mention that the issue template should be used for doing a new release.

**Comment by ffmancera** [comment]
Hi, here is my feedback:

* Currently, we need to group up the tasks. Some of them are trivial like "Click save draft" or "Click on publish release".
* PyPi release should be a task too.

**Comment by pavitra14** [comment]
> * Currently, we need to group up the tasks. Some of them are trivial like "Click save draft" or "Click on publish release".

Like merging them into one?

**Comment by ffmancera** [comment]
> > * Currently, we need to group up the tasks. Some of them are trivial like "Click save draft" or "Click on publish release".
> 
> Like merging them into one?
> 
>     * [ ]  Click `Save draft` and seek for review, Once approved click `Publish release`

In my opinion it would be nice to do something like.

* [ ] Tag the release and save it a s a draft.
* [ ] Verify the release.
```
Instructions to verify it. (Not as a task list)
```
* [ ] Publish the release.

**Comment by pavitra14** [comment]
image
Somewhat like this?

**Comment by ffmancera** [comment]
> image
> Somewhat like this?

Yes, that is quite better. But we need to do that with a lot of tasks. Please, review the document.

**Comment by EdDev** [comment]
I actually use `README.release.md` all the time for the release process.
I do not understand why documentation should be pushed back so deep.

Some of the steps there are not even github related.

Could you please explain the reasoning for moving that documentation?

**Comment by tyll** [comment]
> I actually use `README.release.md` all the time for the release process.
> I do not understand why documentation should be pushed back so deep.
> 
> Some of the steps there are not even github related.
> 
> Could you please explain the reasoning for moving that documentation?

It allows to create a new issue from this template with all the steps as tasks, see for example #916

**Comment by cathay4t** [comment]
Merged. Thanks for the help.

**Pull request merged**